Support Rocky Linux 9

khaledk2 opened the pull request:
I have added a subfolder (rockylinux) to the docker folder, it contains the required files to create a rockylinux 9 docker image. I have installed a rocky Linux version 9.2 on an old laptop (8 GB RAM, i5-8350U CPU) I have created a rocky Linux docker image and tested it on the laptop and it seems to be working fine.

sbesson commented on requirements.txt:
```diff
@@ -20,7 +20,8 @@ numpy==1.21.2
 prompt-toolkit==3.0.20
-psycopg2==2.9.1
+#psycopg2==2.9.1
+psycopg2-binary==2.9.1
```
Is that expected to be a dev dependency only - https://www.psycopg.org/docs/install.html#psycopg-vs-psycopg-binary ?

khaledk2 commented on deployment/docker/rockylinux/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+RUN  dnf install openssl-devel -y
+RUN dnf install libffi-devel -y
+RUN wget https://www.python.org/ftp/python/3.9.0/Python-3.9.0.tgz
+RUN tar xvf Python-3.9.0.tgz
```
I have checked the base docker image (rockylinux:9.0) and it has Python 3.9.10 installed but no pip. So, I have added the installation of pip.

sbesson reviewed (commented):
Comparing the Dockerfile with the one used for CentOS7, it looks like there are additional requirements (`openssl-devel,` libffi-devel`, `libpq-devel`). Is that on purpose?

I assume the mid-term intent is to update https://github.com/ome/omero_search_engine/blob/main/.github/workflows/main.yml#L84 and use Rocky Linux 9 as the base image for the container used by IDR in production? Possibly a topic for an upcoming Monday meeting, it would be useful to discuss the integration timeline into the upcoming IDR releases.

khaledk2 commented:
`libpq-devel` and `python3-devel.x86_64 `are required to install  psycopg2 from source.  `openssl-devel` and `libffi-devel `were required to install psycopg2  from binaries. I have removed the unused requirements, e.g. ones that were used to build Python from the source.
I have modified the workflow to use the Rocky Linux 9 as the base image.

sbesson reviewed (commented):
Thanks @khaledk2. Barring a reduction in the number of statements in the Dockerfile which could reduce the size of the image, I don't see anything of concern.

Next question for you and @jburel is what is the process and the timeline for testing & integrating this PR?

jburel commented on deployment/docker/rockylinux/start_gunicorn_serch_engine.sh:
```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+NAME="omero_search_engine"
+USER root
+PYTHONPATH=~/.pyenv/versions/searchengine/bin/
```
should we go for a location similar to what we have for omero e.g.
``/opt/omero/searchengine/venv3``? instead of homedir

khaledk2 replied:
Sorry, this line was redundant. I have cleaned up the script.

jburel commented on configurations/configuration.py:
```diff
@@ -29,6 +30,20 @@ def load_configuration_variables_from_file(config):
+        try:
+            verify_certs = json.load(config.verify_certs)
+        except Exception as ex:
+            print(str(ex))
```
Do you want to keep the ``print``?

jburel commented on deployment/docker/rockylinux/.dockerignore:
```diff
@@ -0,0 +1,3 @@
+Dockerfile
+.git
+.gitignore
```
need an extra line

khaledk2 replied:
Sorry, I am not sure what you mean
